x-pack/plugins/observability/server/services/slo/summary_cleanup_task.ts

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClient, Logger, SavedObjectsClient } from '@kbn/core/server';
+import {
+  ConcreteTaskInstance,
+  TaskManagerSetupContract,
+  TaskManagerStartContract,
+} from '@kbn/task-manager-plugin/server';
+import { SortResults } from '@elastic/elasticsearch/lib/api/types';
+import { SLO_SUMMARY_DESTINATION_INDEX_PATTERN } from '../../assets/constants';
+import { StoredSLO } from '../../domain/models';
+import { SO_SLO_TYPE } from '../../saved_objects';
+
+export const TASK_TYPE = 'SLO-DEFINITIONS-CLEANUP-TASK';
+
+function findMissingItems(soIds: string[], sloDefIds: string[]) {
+  const missingSloDefIds = sloDefIds.filter((item) => !soIds.includes(item));
+  const missingSOIds = soIds.filter((item) => !sloDefIds.includes(item));
```

Comment:
we need to compare slo id and slo revision.

Also If I follow correctly, here we get two array of ids:
- the first one contains the summary slo id that don't have a saved object (definition) match
- the second one contains the saved object ids (definitions) that don't have a summary match

Am I following correctly?